### PR TITLE
Weekly Patch Release v1.4.8 [full merge, no squash]

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

```bash
gh pr list -s merged -S 'merged:2021-09-14T18:30:00.000Z..2021-09-21T18:30:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "v1.4.x") or (.milestone.title == null)) | [.mergedAt, .url, .mergeCommit.oid, .title] | join(" ")' --limit 100
```

```py
2021-09-20T17:50:41Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9602 2e17b471f865ed76c940b027f19a50176a69af4b Fixing missing URLs
2021-09-17T16:11:19Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9578 8b9d977a08630fa93d0826522ec61c2a9d348fed Update CONTRIBUTING for successful docs build
2021-09-17T09:24:00Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9554 11c93d903dffbf7b90e9a5e512eec34872b09dca Fix `add_argparse_args` raising `TypeError` with Python 3.6
2021-09-16T14:14:38Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9420 982a9560a5066c096a7a28691bb109058fd92ed0 Update notebooks submodule and add tutorial view to docs
2021-09-15T15:44:07Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9522 86c033572da084fd85d4d4c2d637db18d2d431ec [CI] Add Verify Linked Issue
2021-09-15T10:54:17Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9389 08b591ff090328a198c7ec0f71fc4123cbd12b77 Skip reconciliate_processes if used within a cluster environment that creates processes externally
2021-09-15T07:01:42Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9520 e2eb8324613605f921bd951e439f2f2bb54a18ca [bugfix] Resolve example after LightningCLI update
```

Skipped / _notebooks has conflicts. @Borda Any idea what is the right way to resolve this ?
```
2021-09-15T15:44:07Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9522 86c033572da084fd85d4d4c2d637db18d2d431ec [CI] Add Verify Linked Issue
2021-09-15T07:01:42Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9520 e2eb8324613605f921bd951e439f2f2bb54a18ca [bugfix] Resolve example after LightningCLI update
2021-09-16T14:14:38Z https://github.com/PyTorchLightning/pytorch-lightning/pull/9420 982a9560a5066c096a7a28691bb109058fd92ed0 Update notebooks submodule and add tutorial view to docs
```

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_examples/basic_examples/profiler_example.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 from pytorch_lightning.utilities.cli import LightningCLI
 
 DEFAULT_CMD_LINE = (
+    "fit",
```
Comment (carmocca) [inline pl_examples/basic_examples/profiler_example.py:38]: This isn't in the bugfix branch

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,6 +41,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed signature of `Timer.on_train_epoch_end` and `StochasticWeightAveraging.on_train_epoch_end` to prevent unwanted deprecation warnings ([#9347](https://github.com/PyTorchLightning/pytorch-lightning/pull/9347))
 
 
+- Fixed error reporting in DDP process reconciliation when processes are launched by an external agent ([#9389](https://github.com/PyTorchLightning/pytorch-lightning/pull/9389))
+
+
```
Comment (carmocca) [inline CHANGELOG.md:46]: This shouldn't be here

File: .github/workflows/ci_verify_pr_contains_issue.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name: VerifyIssue
+
+on:
+  pull_request:
+    branches: [master]
```
Comment (carmocca) [inline .github/workflows/ci_verify_pr_contains_issue.yaml:5]: Any reason why you picked this too? It won't run anyways

File: pl_examples/basic_examples/profiler_example.py (hunk 1 above)
Comment (tchaton) [reply pl_examples/basic_examples/profiler_example.py:38]: This was tagged as such. Moved it back.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
+## [1.4.8] - 2021-09-21
+
+- Fixed error reporting in DDP process reconciliation when processes are launched by an external agent (#9389)
+- Added PL_RECONCILE_PROCESS environment variable to enable process reconciliation regardless of cluster environment settings (#9389)
+
```
Comment (awaelchli) [inline CHANGELOG.md:11]:
```suggestion
```

**Review by awaelchli — APPROVED** [review]
@tchaton it needs to bump the version in `pl/__about__.py `

**Comment by Borda** [comment]
yeah @tchaton I would not include the 982a9560a5066c096a7a28691bb109058fd92ed0

File: CHANGELOG.md (hunk 2 above)
Comment (tchaton) [reply CHANGELOG.md:46]: What do you mean ? The commit or the changelog ?

Comment (tchaton) [reply CHANGELOG.md:46]: Resolved !

**Comment by tchaton** [comment]
> yeah @tchaton I would not include the [982a956](https://github.com/PyTorchLightning/pytorch-lightning/commit/982a9560a5066c096a7a28691bb109058fd92ed0)

Thanks for telling me. Skipped.

**Comment by tchaton** [comment]
> @tchaton it needs to bump the version in `pl/__about__.py`

Done !

File: CHANGELOG.md (hunk 2 above)
Comment (carmocca) [reply CHANGELOG.md:46]: I mean the changelog entry, it's already above!

File: .github/workflows/ci_verify_pr_contains_issue.yaml (hunk 3 above)
Comment (tchaton) [reply .github/workflows/ci_verify_pr_contains_issue.yaml]: Oh I missed it. Thanks !

**Review by carmocca — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**